### PR TITLE
FIX: Patch Python lower bound on anaconda-client v1.12.0

### PR DESCRIPTION
**Pull request by matthewfeickert**
* Follow up to https://github.com/conda-forge/anaconda-client-feedstock/pull/44
* Helps to resolve https://github.com/Anaconda-Platform/anaconda-client/issues/678

Checklist
* [x] Ran `python show_diff.py` and posted the output as part of the PR.

```console
$ docker run --rm -ti -v $PWD:/work -w /work mambaorg/micromamba:1.4.9-bullseye-slim
(base) mambauser@a147f8088436:/work$ micromamba create --yes --name debug --channel conda-forge python tqdm requests packaging license-expression conda-build
(base) mambauser@a147f8088436:/work$ micromamba activate debug
(debug) mambauser@a147f8088436:/work$ python recipe/show_diff.py 
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata.json.bz2
noarch::anaconda-client-1.12.0-pyhd8ed1ab_0.conda
-    "python >=3.6",
+    "python >=3.8",
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata.json.bz2
```

* [x] Modifications won't affect packages built in the future. 
   - This conditions on the build number, not the datetime.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@jakirkham, @goanpeca, @beckermr as you are listed as being the `anaconda-client` feedstock maintainers please review this.

**Comment by beckermr** [comment]
The patch appears to remove the python-dateutil dep. Please fix.

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -2389,6 +2388,15 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
                             record["depends"],
                             record,
                         )
+            # https://github.com/conda-forge/anaconda-client-feedstock/pull/44
+            # https://github.com/Anaconda-Platform/anaconda-client/issues/678
+            if (
+                pkg_resources.parse_version(record["version"])
+                == pkg_resources.parse_version("1.12.0")
+            ) and record["build_number"] == 0:
+                python_pinning = [x for x in record["depends"] if x.startswith("python")]
```
Comment (beckermr) [inline recipe/gen_patch_json.py:2397]:
```suggestion
                python_pinning = [x for x in record["depends"] if x.startswith("python ")]
```

**Review by beckermr — CHANGES_REQUESTED** [review]
I think this will fix it.

**Comment by beckermr** [comment]
Pls post a new diff. Thank you!

**Comment by matthewfeickert** [comment]
> I think this will fix it.

Yes, seems so. I'm going to rebase this to avoid noise in the commit history. 

> Pls post a new diff. Thank you!

Still running, as I'm on a slow 4G connection in a remote area currently, but looks correct now:

```console
(debug) root@30cca0b0ce64:/debug# python show_diff.py 
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata.json.bz2
noarch::anaconda-client-1.12.0-pyhd8ed1ab_0.conda
-    "python >=3.6",
+    "python >=3.8",
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata.json.bz2
```

**Comment by jakirkham** [comment]
Tried to re-request a review. Sorry if that wound up being messy

**Comment by matthewfeickert** [comment]
Yeah, I had already done that. :/

**Review by beckermr — APPROVED** [review]

**Comment by matthewfeickert** [comment]
Thanks very much @beckermr and @jakirkham.

**Comment by jakirkham** [comment]
> Yeah, I had already done that. :/ But no worries.

The challenges of using GitHub mobile 😅

Anyways still much improved over what use to be possible

**Comment by matthewfeickert** [comment]
> The challenges of using GitHub mobile

Haha, 100% feel that. :) I appreciate you just trying to move things along, so thank you very much and thank you for pointing me in the right directions to get this fixed! :pray: 

> Anyways still much improved over what use to be possible

Everytime I make a PR on a plane now from my phone it feels like pure magic. :sparkles:

**Comment by jakirkham** [comment]
Right back at you :)

Interesting are using GitHub Mobile to create PRs or something else like Working Copy?

**Comment by matthewfeickert** [comment]
> Interesting are using GitHub Mobile to create PRs or something else like Working Copy?

Just GitHub Mobile + GitHub the website from browser on phone (to check CI). Not great, but possible! I didn't know about Working Copy, but I'll check that out.

**Pull request merged**